# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: js/src/vm/HelperThreads.h
# Commit: 68e6b05b17b7
# Full Hash: 68e6b05b17b7e32496b5519f0b0fd9e99e10b3a0
# Author: Jon Coppeard <jcoppeard@mozilla.com>
# Date: 2020-10-28 09:24:21
# Description:
#   Bug 1672760 - Add an API to wait for parse tasks to complete without cancelling them r=sfink
#   
#   The problem here was cancelling parse tasks without the browser's knowledge (I
#   didn't realise that the cancel method did anything beyond waiting).
#   
# ==============================================================================

diff -r b55c94de3ec7 -r 68e6b05b17b7 js/src/vm/HelperThreads.h
--- a/js/src/vm/HelperThreads.h	Tue Oct 27 15:24:54 2020 +0000
+++ b/js/src/vm/HelperThreads.h	Tue Oct 27 14:10:27 2020 +0000
@@ -177,6 +177,12 @@
 bool HasOffThreadIonCompile(JS::Realm* realm);
 #endif
 
+/*
+ * Wait for all scheduled, in progress or finished parse tasks for the runtime
+ * to complete.
+ */
+void WaitForOffThreadParses(JSRuntime* runtime);
+
 /* Cancel all scheduled, in progress or finished parses for runtime. */
 void CancelOffThreadParses(JSRuntime* runtime);
 